bundles/org.openhab.binding.adorne/src/main/java/org/openhab/binding/adorne/internal/handler/AdorneSwitchHandler.java

```diff
@@ -0,0 +1,184 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.adorne.internal.handler;
+
+import static org.openhab.binding.adorne.internal.AdorneBindingConstants.CHANNEL_POWER;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.adorne.internal.configuration.AdorneSwitchConfiguration;
+import org.openhab.binding.adorne.internal.hub.AdorneHubController;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AdorneSwitchHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Mark Theiding - Initial contribution
+ */
+@NonNullByDefault
+public class AdorneSwitchHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(AdorneSwitchHandler.class);
+
+    /**
+     * The zone ID that represents this {@link AdorneSwitchHandler}'s thing
+     */
+    protected int zoneId;
+
+    public AdorneSwitchHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * Handles refresh and on/off commands for channel
+     * {@link org.openhab.binding.adorne.internal.AdorneBindingConstants#CHANNEL_POWER}
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("handleCommand (channelUID:{} command:{}", channelUID, command);
+        try {
+            if (channelUID.getId().equals(CHANNEL_POWER)) {
+                if (command.equals(OnOffType.ON) || command.equals(OnOffType.OFF) || command instanceof RefreshType) {
```

Comment:
```suggestion
                if (command.equals(OnOffType.ON) || command.equals(OnOffType.OFF)) {
```

and then later `} else (if command instanceof RefreshType`

Reply:
Yikes, how did I come up with that one? Good catch. Done